src/main/java/duke/main/Duke.java

```diff
@@ -0,0 +1,166 @@
+package duke.main;
+
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.TaskException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+
+public class Duke {
+    private final Storage storage;
+    private final Ui ui;
+    private TaskList tasks;
+
+    /**
+     * Create duke object.
+     * @param filePath Specify storage file path.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = storage.load();
+        } catch (FileNotFoundException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Adds a new task depending on type and append in to tasks.
+     *
+     * @param str       the whole input string, split.
+     * @throws TaskException if no details when adding.
+     */
+    private void addTask(String[] str) throws TaskException {
+        if (str[0].compareTo("deadline") == 0) {
```

Comment:
The code you have uses deep nesting code. Perhaps you could use a switch case, and abstract out the if else clauses for better readability? 
I've noticed this in other parts of your code as well.